src/app/me/tickets/ticket-transfer.dialog.tsx

```diff
@@ -0,0 +1,152 @@
+import { zodResolver } from "@hookform/resolvers/zod"
+import copy from "copy-to-clipboard"
+import { Clipboard, SendHorizontal } from "lucide-react"
+import { type ReactNode, useState } from "react"
+import { useForm } from "react-hook-form"
+import { toast } from "sonner"
+import { z } from "zod"
+
+import { Button } from "@/components/ui/button"
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog"
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form"
+import { Input } from "@/components/ui/input"
+import doMailto from "@/lib/do-mailto"
+import { api } from "@/trpc/react"
+
+export interface TransferTicketsDialogProps {
+  children: ReactNode
+  ticketIds: string[]
+  onDone: () => void
+}
+
+export default function TransferTicketsDialog(
+  props: TransferTicketsDialogProps,
+) {
+  const [opened, setOpened] = useState(false)
+  const [loading, setLoading] = useState(false)
+  const apiTransferTicket = api.ticket.createTransfer.useMutation()
+
+  const formSchema = z.object({
+    email: z.string().email(),
+  })
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      email: "",
+    },
+  })
+
+  async function onSubmit(values: z.infer<typeof formSchema>) {
+    setLoading(true)
+    const transferIds = await Promise.all(
+      props.ticketIds.map((ticketId) =>
+        apiTransferTicket.mutateAsync({ ticketId }),
+      ),
+    )
+
+    const shareLink = `${window.location.origin}/me/tickets?redeemCode=${transferIds.join(",")}`
+
+    doMailto(
+      values.email,
+      "You have received tickets:)",
+      `Hi\n\nYou have received tickets on Luttaka.\n\n You can redeem them by visiting the following link:\n ${shareLink}`,
+    )
```

Comment:
This would be cool to implement a mail template generation. But that is far beyond the scope of this issue :D